extensions/data-workspace/src/main.ts

```diff
@@ -47,7 +49,12 @@ export function activate(context: vscode.ExtensionContext): WorkspaceService {
 		await workspaceService.removeProject(vscode.Uri.file(treeItem.element.project.projectFilePath));
 	}));
 
-	return workspaceService;
+	const getProjectsInWorkspace = (): vscode.Uri[] => {
```

Comment:
having a class implementing the interface is to reduce the code you have to write in the main.ts, let's move these to the dataworkspaceExtension class.